packages/babel-plugin/src/utils/css-map.ts

```diff
@@ -17,8 +17,18 @@ export const getKeyValue = (key: ObjectKeyWithLiteralValue): string => {
   throw new Error(`Expected an identifier or a string literal, got type ${(key as any).type}`);
 };
 
-export const isAtRule = (key: ObjectKeyWithLiteralValue): key is ObjectKeyWithLiteralValue =>
-  getKeyValue(key).startsWith('@');
+export const isAtRuleObject = (
+  key: ObjectKeyWithLiteralValue
+): key is ObjectKeyWithLiteralValue => {
+  const keyValue = getKeyValue(key);
+  if (keyValue.includes(' ')) {
+    // Anything that includes a space character won't have an at rule "object".
```

Comment:
not quite sure i'm understanding this correctly... what is the definition for an "at rule object" here?

`'@media (max-width: 256px)': { ... }` has a few space characters inside the media query - we would want it to be true for that right?

Reply:
Maybe the internal code could use a refactor, but basically for cssMap anything that starts with an `@` is assumed to be a nested at rule declaration, which then has its styles flattened prior to being resolved.

```
const styles = cssMap({
  nested: { '@media': { '(min-width: 30rem)': { color: 'red' } } },
});
```

This is no longer true however as we want top level declarations to be valid.

```
const styles = cssMap({
  topLevel: { '@media (min-width: 30rem)': { color: 'green' } },
});
```

An 'easy' way to make that happen is: if it has a space in it it's not something we're interested in flattening. We could flip it and only flatten it if it's a specific value (like at media, at supports, and so on).

Reply:
Unfortunately I'd say `@media(…)`, eg. `@supports(display:flex)` are valid at rules without spaces.

Perhaps `return /^@[a-z]+$/i.test(keyValue);`

Reply:
Yes but that wouldn't be supported by cssMap API (before and after this PR).